### DISCUSSION TITLE
Question on inference library

### DISCUSSION THREAD

**Discussion by haoyanc** [KevinMusgrave/pytorch-metric-learning#455] [Q&A]
Hi, i have recently trained a model using the triplet margin loss notebook as baseline reference with my custom dataset. How does the result from the is_match function different from the accuracy calculated from the triplet loss notebook. The is_match function seems to be performing worse when trying to match data of the same labels.

**Comment by KevinMusgrave** [comment]
The accuracy calculation uses k-nearest-neighbors, whereas ```is_match``` uses thresholding with distance or similarity.

For example:
- k-nearest-neighbors with k = 1: for each image, find the 1st closest neighbor in all of the training data, and if it's the same class, then give this image a score of 1.
- ```is_match```: given two images, if the distance between them is less than the threshold, then it's a match.
